alembic/versions/21dcbd2fac3f_add_smart_routing.py

```diff
@@ -0,0 +1,74 @@
+    sa.Column('project_id', postgresql.UUID(as_uuid=True), nullable=True),
+    sa.Column('strategy_id', postgresql.UUID(as_uuid=True), nullable=True),
+    sa.Column('created_by', postgresql.UUID(as_uuid=True), nullable=True),
+    sa.Column('created_at', sa.DateTime(), nullable=True),
+    sa.Column('field', sa.String(), nullable=True),
+    sa.Column('description', sa.String(), nullable=True),
+    sa.Column('is_input', sa.Boolean(), nullable=True),
+    sa.ForeignKeyConstraint(['created_by'], ['user.id'], ondelete='SET NULL'),
+    sa.ForeignKeyConstraint(['project_id'], ['cognition.project.id'], ondelete='CASCADE'),
+    sa.ForeignKeyConstraint(['strategy_id'], ['cognition.strategy.id'], ondelete='CASCADE'),
+    sa.PrimaryKeyConstraint('id'),
+    schema='cognition'
+    )
+    op.create_index(op.f('ix_cognition_strategy_requirement_created_by'), 'strategy_requirement', ['created_by'], unique=False, schema='cognition')
+    op.create_index(op.f('ix_cognition_strategy_requirement_project_id'), 'strategy_requirement', ['project_id'], unique=False, schema='cognition')
+    op.create_index(op.f('ix_cognition_strategy_requirement_strategy_id'), 'strategy_requirement', ['strategy_id'], unique=False, schema='cognition')
+    op.create_table('strategy_requirement_mapping_option',
+    sa.Column('id', postgresql.UUID(as_uuid=True), nullable=False),
+    sa.Column('project_id', postgresql.UUID(as_uuid=True), nullable=True),
+    sa.Column('strategy_requirement_id', postgresql.UUID(as_uuid=True), nullable=True),
+    sa.Column('created_by', postgresql.UUID(as_uuid=True), nullable=True),
+    sa.Column('created_at', sa.DateTime(), nullable=True),
+    sa.Column('value', sa.String(), nullable=True),
+    sa.ForeignKeyConstraint(['created_by'], ['user.id'], ondelete='SET NULL'),
+    sa.ForeignKeyConstraint(['project_id'], ['cognition.project.id'], ondelete='CASCADE'),
+    sa.ForeignKeyConstraint(['strategy_requirement_id'], ['cognition.strategy_requirement.id'], ondelete='CASCADE'),
+    sa.PrimaryKeyConstraint('id'),
+    schema='cognition'
+    )
+    op.create_index(op.f('ix_cognition_strategy_requirement_mapping_option_created_by'), 'strategy_requirement_mapping_option', ['created_by'], unique=False, schema='cognition')
+    op.create_index(op.f('ix_cognition_strategy_requirement_mapping_option_project_id'), 'strategy_requirement_mapping_option', ['project_id'], unique=False, schema='cognition')
+    op.create_index(op.f('ix_cognition_strategy_requirement_mapping_option_strategy_requirement_id'), 'strategy_requirement_mapping_option', ['strategy_requirement_id'], unique=False, schema='cognition')
+    op.add_column('pipeline_logs', sa.Column('iteration_number', sa.Integer(), nullable=True), schema='cognition')
+    op.add_column('project', sa.Column('operator_routing_config', sa.JSON(), nullable=True), schema='cognition')
+    op.drop_column('project', 'operator_routing_source_code', schema='cognition')
```

Comment:
this will destroy all existing router code! (add & drop for a rename). In normal databases this is usually an ALTER statement to ensure the integrity of the data. Not sure if alembic has this worct case raw sql

Reply:
as Jens mentioned, this change corrupted the DEV database via K8: Test workflow. I manually restored the DB in a working state. @JWittmeyer we may need to rethink how alembic migrations affect the DB during the K8: Test workflow. I will send over some ideas via IM.

> [!CAUTION]
> Please do not push new changes to this branch unless they include the alembic fix, otherwise they will break the DB again